pkg/homedir/homedir_unix.go

```diff
@@ -133,6 +135,33 @@ func GetConfigHome() (string, error) {
 	return rootlessConfigHomeDir, rootlessConfigHomeDirError
 }
 
+// checkBootId checks if the boot_id of the system has changed since the last time the runtime directory was created.
+func checkBootId(path string) error {
+	systemBootId, err := os.ReadFile("/proc/sys/kernel/random/boot_id")
+	if err != nil {
+		if errors.Is(err, os.ErrNotExist) {
+			// The system does not have a boot_id, so ignore the check
+			return nil
+		}
+		return err
+	}
+
+	bootIdFile := filepath.Join(path, "bootid")
+	bootId, err := os.ReadFile(bootIdFile)
+	if err != nil {
+		// If the file does not exist, create it
+		if errors.Is(err, os.ErrNotExist) {
+			return ioutils.AtomicWriteFile(bootIdFile, systemBootId, 0o600)
+		}
+		return err
+	}
+
+	if string(bootId) != string(systemBootId) {
```

Comment:
(Absolutely non-blocking: `bytes.Compare` expresses the same thing without implying a copy, and is clearer about the intent (we don’t care whether the value is valid text). This works fine as is, and it is not performance-sensitive.)